zap/src/main/java/org/zaproxy/zap/session/ScriptBasedSessionManagementMethodType.java

```diff
@@ -0,0 +1,728 @@
+                    this,
+                    errorMessage,
+                    Constant.messages.getString("session.method.script.dialog.error.title"),
+                    JOptionPane.ERROR_MESSAGE);
+            this.loadedScript = null;
+            this.scriptsComboBox.setSelectedItem(null);
+            this.dynamicFieldsPanel = null;
+            this.dynamicContentPanel.removeAll();
+            this.dynamicContentPanel.add(new JLabel(LABEL_NOT_LOADED), BorderLayout.CENTER);
+            this.dynamicContentPanel.revalidate();
+        }
+
+        @Override
+        public void bindMethod(SessionManagementMethod method)
+                throws UnsupportedSessionManagementMethodException {
+            this.method = (ScriptBasedSessionManagementMethod) method;
+
+            // Make sure the list of scripts is refreshed
+            List<ScriptWrapper> scripts = getScriptsExtension().getScripts(SCRIPT_TYPE_SESSION);
+            DefaultComboBoxModel<ScriptWrapper> model =
+                    new DefaultComboBoxModel<>(scripts.toArray(new ScriptWrapper[scripts.size()]));
+            this.scriptsComboBox.setModel(model);
+            this.scriptsComboBox.setSelectedItem(this.method.script);
+            this.loadScriptButton.setEnabled(this.method.script != null);
+
+            // Load the selected script, if any
+            if (this.method.script != null) {
+                loadScript(this.method.script, false);
+                if (this.dynamicFieldsPanel != null) {
+                    this.dynamicFieldsPanel.bindFieldValues(this.method.paramValues);
+                }
+            }
+        }
+
+        @Override
+        public SessionManagementMethod getMethod() {
+            return this.method;
+        }
+    }
+
+    /**
+     * A renderer for properly displaying the name of a {@link ScriptWrapper} in a ComboBox and
+     * putting emphasis on loaded script.
+     */
+    private static class ScriptWrapperRenderer extends BasicComboBoxRenderer {
+        private static final long serialVersionUID = 3654541772447187317L;
+        private static final Border BORDER = new EmptyBorder(2, 3, 3, 3);
+        private ScriptBasedSessionManagementMethodOptionsPanel panel;
+
+        public ScriptWrapperRenderer(ScriptBasedSessionManagementMethodOptionsPanel panel) {
+            super();
+            this.panel = panel;
+        }
+
+        @Override
+        @SuppressWarnings("rawtypes")
+        public Component getListCellRendererComponent(
+                JList list, Object value, int index, boolean isSelected, boolean cellHasFocus) {
+            super.getListCellRendererComponent(list, value, index, isSelected, cellHasFocus);
+            if (value != null) {
+                setBorder(BORDER);
+                ScriptWrapper item = (ScriptWrapper) value;
+                if (panel.loadedScript == item)
+                    setText("<html><b>" + item.getName() + " (loaded)</b></html>");
```

Comment:
Is this XSS'able? (I guess it'd be self XSS but....)

Reply:
Good point, also ripped off from https://github.com/zaproxy/zaproxy/blob/develop/zap/src/main/java/org/zaproxy/zap/authentication/ScriptBasedAuthenticationMethodType.java#L575

Reply:
Have also fixed that in ScriptBasedAuthenticationMethodType.java